portlayer/attach/connector.go

```diff
@@ -137,22 +137,30 @@ func (c *Connector) Remove(id string) {
 
 // takes the base connection, determines the ID of the source and stashes it in the map
 func (c *Connector) processIncoming(conn net.Conn) {
+	var err error
+	defer func() {
+		if err != nil && conn != nil {
```

Comment:
If err == nil and conn is not nil, we dont need to do anything.  if err is nil and conn is nil, again we don't need to do anything.